schema/derive/src/lib.rs

```diff
@@ -1,54 +1,36 @@
 //! Crate with derive `IntoSchema` macro
 
 #![allow(clippy::arithmetic_side_effects)]
-
-use proc_macro::TokenStream;
-use proc_macro2::{Ident, Span, TokenStream as TokenStream2};
-use quote::quote;
-use syn::{
-    parse::{Parse, ParseStream, Result},
-    parse_macro_input, parse_quote,
-    spanned::Spanned,
-    Attribute, Data, DataEnum, DataStruct, DeriveInput, Expr, Field, Fields, FieldsNamed,
-    FieldsUnnamed, GenericParam, Generics, LitStr, Meta, NestedMeta, Type,
-};
+// darling-generated code triggers this lint
+#![allow(clippy::option_if_let_else)]
+// it's fine to have some functions return Result without returning Err
+// this allows some functions to have more uniform signatures
+#![allow(clippy::unnecessary_wraps)]
```

Comment:
No. 

1) You are lying to the user of your lib. 
2) While sometimes the return of the result is required because of a function signature, you don't have it in a macro crate. You are also extremely unlikely to have it in a macro crate all things considered. 
3) If it does happen you return `Result<Thing, Infallible>` customarily, which at least helps the user see that it is not a fallible operation. 
4) If you **did** in fact have the need to return a `Result` because of trait bounds (see point 2), you could do so by taking an infallible function and returning a closure that wraps it in an `Ok()` after evaluation. It's much neater in Haskell than in Rust, true, but you still get the benefit of removing these instances if and when you need them, without actually having to refactor the function signature.

Reply:
I like when most of my functions return `Result` (with a universal error type), mostly for "symmetry" and for ease of adding error cases in the future (no need to rewrite the signature and call sites), but I am not going to die on this hill.

Reply:
Rust did not go with the symmetrical solution of exceptions (panics) for error handling, expressly for this purpose. 

If the function suddenly becomes fallible, the change in the signature requires some downstream modification. If you don't change the signature but make a previously infallible function falllible, you no longer have a nice compilation error to tell you what needs to be done. A function which didn't fail previously suddenly failing now is an API breaking change.

Reply:
The only reason I understand is future changes. Although for that library developers basically use complex wrappers around real errors, use `#[non_exhaustive]` and etc.

**But** you're not writing extremely portable library for hundreds of users. You're writing our internal crate, it's even not in the crates.io. So I agree with Aleksandr.